share/completions/zfs.fish

```diff
@@ -10,7 +10,7 @@
 # - for the program command, complete the script to be executed
 # - for commands accepting several arguments of different types, propose arguments in the right order: for get, once the ZFS parameters have been given, only propose datasets
 
-set OS ""
+set -lx OS ""
```

Comment:
Yeah, this will have to be done via "--inherit-variable", so

```fish
function __fish_zfs_list_permissions -V OS
```

$OS is used in that one function and outside of functions, so that seems to be the simplest way to keep computing it only once.